### PR TITLE
JAMES-2731 Improve MailboxPath logging

### PR DESCRIPTION
**Pull request by chibenwa**
getFullName does not include username, so we miss some debugging information.

Using MailboxPath::asString is the right alternative.

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java [hunk 1]
```diff
@@ -178,14 +178,4 @@ public final int hashCode() {
         return Objects.hash(namespace, user, name);
     }
 
-    /**
-     * Return the full name of the {@link MailboxPath}, which is constructed via the {@link #namespace} and {@link #name}
-     * 
-     * @param delimiter
-     * @return fullName
-     */
-    public String getFullName(char delimiter) {
```
Comment (rouazana) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java]: no need to deprecate this at first?

Comment (chibenwa) [reply mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java]: Could be if you prefer. IMO we never refrained ourselves from breaking changes within the mailbox/api so far...

Comment (rouazana) [reply mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java]: Yes, I'm not sure, but MailboxPath is a very broadly used one...

Comment (chibenwa) [reply mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java]: Ok for only deprecating this...

Comment (mbaechler) [reply mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java]: I would have remove it too

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**